### PR TITLE
Add genomes to v4 validity checks

### PR DESCRIPTION
**Pull request by mike-w-wilson**
This adds genomes to the v4 validity check script. There is a hardcoded test path and commented out code that I'll delete and restore once this is ready but leaving this here for easy testing.

### CODE REVIEW

File: gnomad_qc/v4/create_release/validate_and_export_vcf.py [hunk 1]
```diff
@@ -53,26 +62,32 @@
...
-)
+    ],
+    "genomes": REGION_FLAG_FIELDS,
+}
 
 # Remove original alleles for containing non-releasable alleles
 ALLELE_TYPE_FIELDS = deepcopy(ALLELE_TYPE_FIELDS)
 MISSING_ALLELE_TYPE_FIELDS = ["original_alleles"]
 ALLELE_TYPE_FIELDS = remove_fields_from_constant(
     ALLELE_TYPE_FIELDS, MISSING_ALLELE_TYPE_FIELDS
```
Comment (jkgoodrich) [inline gnomad_qc/v4/create_release/validate_and_export_vcf.py:84]:
```suggestion
ALLELE_TYPE_FIELDS = remove_fields_from_constant(
    ALLELE_TYPE_FIELDS, ["original_alleles"]
```

File: gnomad_qc/v4/create_release/validate_and_export_vcf.py [hunk 2]
```diff
@@ -84,15 +99,29 @@
...
-SUBSETS = deepcopy(SUBSETS["v4"])
+GENOME_SUBSETS_TO_DROP = remove_fields_from_constant(
+    deepcopy(SUBSETS["v3"]), ["hgdp", "tgp"]
+)
+SUBSETS = {
+    "exomes": deepcopy(SUBSETS["v4"]),
+    "genomes": remove_fields_from_constant(
+        deepcopy(SUBSETS["v3"]), GENOME_SUBSETS_TO_DROP
+    ),
+}
```
Comment (jkgoodrich) [inline gnomad_qc/v4/create_release/validate_and_export_vcf.py:110]:
```suggestion
SUBSETS = {
    "exomes": deepcopy(SUBSETS["v4"]),
    "genomes": remove_fields_from_constant(
        deepcopy(SUBSETS["v3"]), ["hgdp", "tgp"]
    ),
}
```

File: gnomad_qc/v4/create_release/validate_and_export_vcf.py [hunk 3]
```diff
@@ -12,7 +12,13 @@
     pprint_global_anns,
     validate_release_t,
 )
-from gnomad.resources.grch38.gnomad import POPS, SUBSETS
+from gnomad.resources.grch38.gnomad import (
+    HGDP_POPS,
+    POPS,
+    SUBSETS,
+    TGP_POP_NAMES,
```
Comment (jkgoodrich) [inline gnomad_qc/v4/create_release/validate_and_export_vcf.py:19]: I don't think this is used

File: gnomad_qc/v4/create_release/validate_and_export_vcf.py [hunk 4]
```diff
@@ -12,7 +12,13 @@
     pprint_global_anns,
     validate_release_t,
 )
-from gnomad.resources.grch38.gnomad import POPS, SUBSETS
+from gnomad.resources.grch38.gnomad import (
+    HGDP_POPS,
```
Comment (jkgoodrich) [inline gnomad_qc/v4/create_release/validate_and_export_vcf.py:16]: I think HGDP_POPS (and for that matter HGDP_POP_NAMES) https://github.com/broadinstitute/gnomad_methods/blob/0d2e71f93d5cfceebfad2ab538e47d8457d57d6d/gnomad/resources/grch38/gnomad.py#L189 need to be updated @KoalaQin ?

**Review by jkgoodrich — CHANGES_REQUESTED** [review]
Only some small things. Also, we have a check for joint fafmax, are we supposed to have one for fafmax too?

**Comment by mike-w-wilson** [comment]
Yes -- should have been one for fafmax, just added!

File: gnomad_qc/v4/create_release/validate_and_export_vcf.py [hunk 5]
```diff
@@ -133,8 +158,16 @@ def get_export_resources(
...
-        output_resources={"validated_ht": validated_release_ht(test=test)},
+        input_resources={
+            "create_release_sites_ht.py": {
+                "release_ht": hl.read_table(
+                    "gs://gnomad-tmp/gnomad.exomes.v4.0.qc_data/release/gnomad.genomes.sites.test.updated_101923.ht"
+                )  # release_sites(data_type=data_type) #TODO: Restore once PR is ready for merge
+            }
+        },
+        output_resources={
+            "validated_ht": validated_release_ht(test=test, data_type=data_type)
```
Comment (jkgoodrich) [inline gnomad_qc/v4/create_release/validate_and_export_vcf.py:169]: I just noticed `validated_release_ht` has an overwrite parameter that can me removed

**Review by jkgoodrich — COMMENTED** [review]
just one thing and removal of TODOs then this is good to go. @KoalaQin can you look into edits for the `HGDP_POPS` and `HGDP_POP_NAMES` in `gnomad.resources.grch38.gnomad` ?

**Review by jkgoodrich — APPROVED** [review]
LGTM!

**Comment by KoalaQin** [comment]
> just one thing and removal of TODOs then this is good to go. @KoalaQin can you look into edits for the `HGDP_POPS` and `HGDP_POP_NAMES` in `gnomad.resources.grch38.gnomad` ?

@jkgoodrich, I updated HGDP_POPs in the PR I tagged you, but I can't find `HGDP_POP_NAMES`.

**Pull request merged**